br/pkg/lightning/backend/local/local_test.go

```diff
@@ -837,49 +821,119 @@ func TestMergeSSTsDuplicated(t *testing.T) {
 
 type mockPdClient struct {
 	pd.Client
-	stores []*metapb.Store
+	stores  []*metapb.Store
+	regions []*pd.Region
 }
 
 func (c *mockPdClient) GetAllStores(ctx context.Context, opts ...pd.GetStoreOption) ([]*metapb.Store, error) {
 	return c.stores, nil
 }
 
+func (c *mockPdClient) ScanRegions(ctx context.Context, key, endKey []byte, limit int) ([]*pd.Region, error) {
+	return c.regions, nil
+}
+
 type mockGrpcErr struct{}
 
 func (e mockGrpcErr) GRPCStatus() *status.Status {
-	return status.New(codes.Unimplemented, "unimplmented")
+	return status.New(codes.Unimplemented, "unimplemented")
 }
 
 func (e mockGrpcErr) Error() string {
-	return "unimplmented"
+	return "unimplemented"
 }
 
 type mockImportClient struct {
 	sst.ImportSSTClient
 	store              *metapb.Store
+	resp               *sst.IngestResponse
 	err                error
 	retry              int
 	cnt                int
 	multiIngestCheckFn func(s *metapb.Store) bool
+	apiInvokeRecorder  map[string][]uint64
+}
+
+func newMockImportClient() *mockImportClient {
+	return &mockImportClient{
+		multiIngestCheckFn: func(s *metapb.Store) bool {
+			return true
+		},
+	}
 }
 
 func (c *mockImportClient) MultiIngest(context.Context, *sst.MultiIngestRequest, ...grpc.CallOption) (*sst.IngestResponse, error) {
 	defer func() {
 		c.cnt++
 	}()
-	if c.cnt < c.retry && c.err != nil {
-		return nil, c.err
+	if c.apiInvokeRecorder != nil {
+		c.apiInvokeRecorder["MultiIngest"] = append(c.apiInvokeRecorder["MultiIngest"], c.store.GetId())
+	}
+	if c.cnt < c.retry && (c.err != nil || c.resp != nil) {
+		return c.resp, c.err
 	}
 
 	if !c.multiIngestCheckFn(c.store) {
 		return nil, mockGrpcErr{}
 	}
-	return nil, nil
+	return &sst.IngestResponse{}, nil
+}
+
+type mockWriteClient struct {
+	writeResp *sst.WriteResponse
+}
+
+func (m mockWriteClient) Send(request *sst.WriteRequest) error {
+	return nil
+}
+
+func (m mockWriteClient) CloseAndRecv() (*sst.WriteResponse, error) {
+	return m.writeResp, nil
+}
+
+func (m mockWriteClient) Header() (metadata.MD, error) {
+	//TODO implement me
+	panic("implement me")
```

Comment:
We can embed `sst.ImportSST_WriteClient` inside `mockWriteClient` so that we don't need to add such unimplemented methods.